### PR TITLE
Add quickstart example for task queue functions.

### PR DESCRIPTION
**Pull request by taeold**
NOTE: Requires custom build of the admin SDK

### CODE REVIEW

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for t`he specific language governing permissions and
+ * limitations under the License.
+ */
+// [START import]
+"use strict";
+const path = require("path");
+const fetch = require("node-fetch");
+const functions = require("firebase-functions");
```
Comment (thatfiredev) [inline quickstarts/taskqueues-backup-images/functions/index.js:20]:
```suggestion
const functions = require("firebase-functions/v2");
```

Comment (thatfiredev) [reply quickstarts/taskqueues-backup-images/functions/index.js:20]: Btw, v2 also supports modular imports so maybe we should demonstrate that too

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+initializeApp();
+// [END import]
+
+
+// [START backupImages]
+/**
+ * Grabs Astronomy Photo of the Day (APOD) using NASA's API.
+ *
+ */
+exports.backupApod = functions
```
Comment (thatfiredev) [inline quickstarts/taskqueues-backup-images/functions/index.js:35]: v2 only supports lowercase function names.

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      return;
+    });
+
+// [START backupSetup]
+const BACKUP_START_DATE = new Date("1995-06-17");
+const BACKUP_COUNT = 50; // TODO: Update this to 9000
+const HOURLY_BATCH_SIZE = 10; // TODO: Update this to 500
+// [START backupSetup]
+
+exports.enqueueBackupTasks = functions.https.onRequest(
```
Comment (thatfiredev) [inline quickstarts/taskqueues-backup-images/functions/index.js:112]: same comment as above

File: quickstarts/taskqueues-backup-images/functions/index.js (hunk 2 above)
Comment (taeold) [reply quickstarts/taskqueues-backup-images/functions/index.js:35]: technically, this is v1 task queue functions. Should I be just targeting the main branch in that case?

File: quickstarts/taskqueues-backup-images/functions/index.js (hunk 1 above)
Comment (taeold) [reply quickstarts/taskqueues-backup-images/functions/index.js:20]: TQ functions works slightly better w/ v1 functions for now, as long as v2 functions don't offer deterministic url.

File: quickstarts/taskqueues-backup-images/functions/index.js (hunk 2 above)
Comment (thatfiredev) [reply quickstarts/taskqueues-backup-images/functions/index.js:35]: Oh, my bad, I didn't know this was available in v1. I'll wait for Jeff to confirm which branch to target.

**Comment by jhuleatt** [comment]
There are just under 10,000 astronomy pictures of the day ([archive](https://apod.nasa.gov/apod/archivepixFull.html)). So theoretically we could slurp down the whole thing in two hours while still respecting the 9k/hour rate limit of the NASA API. That's probably ok, just wanted to point it out

File: quickstarts/taskqueues-backup-images/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# Quickstart: Backup images using a Task Queue Function
```
Comment (egilmorez) [inline quickstarts/taskqueues-backup-images/README.md:1]: Suggest "Back up"  (the verb, not the noun)

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+const {initializeApp} = require("firebase-admin/app");
+const {getFunctions} = require("firebase-admin/functions");
+const {getStorage} = require("firebase-admin/storage");
+const logger = functions.logger;
+const HttpsError = functions.https.HttpsError;
+initializeApp();
+// [END import]
+
+
+// [START backupImages]
```
Comment (egilmorez) [inline quickstarts/taskqueues-backup-images/functions/index.js:30]: Global comment: we only have to tag elements that we want to actually put in the docs, as snippets.  

So we don't need tags for each and every action in the sample.

**Review by egilmorez — COMMENTED** [review]
Some initial thoughts Daniel, thanks!

I figure the sample can be as wonderful and elaborate as you want, but the tagged parts should be somewhat more limited and directed toward doc usage.  Make sense?

File: quickstarts/taskqueues-backup-images/functions/index.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Grabs Astronomy Photo of the Day (APOD) using NASA's API.
+ *
+ */
+exports.backupApod = functions
+    // [START securelyStoreAPIKey]
+    .runWith( {secrets: ["NASA_API_KEY"]})
+    // [END securelyStoreAPIKey]
```
Comment (jhuleatt) [inline quickstarts/taskqueues-backup-images/functions/index.js:38]: FYI @egilmorez we can use this snippet for Secret Manager docs

**Comment by taeold** [comment]
Closing in favor of https://github.com/firebase/functions-samples/pull/971

**Pull request closed without merging**